### PR TITLE
Fix `getGarbageLen` to retrun correct size

### LINKED ISSUE #688: `getGarbageLen()` does not match the actual number of elements for garbage collection
**What happened**:

The `getGarbageLen()` function iterates through elements in `root.removedElementSetByCreatedAt`, counting the descendants' numbers. This causes elements to be counted multiple times, leading to a difference between the count provided by `getGarbageLen()` and the actual number of elements being garbage collected.

**What you expected to happen**:

The counts by `getGarbageLen()` and the actual number of elements being garbage collected should match.

**How to reproduce it (as minimally and precisely as possible)**:

Execute the following test code in the yorkie-js-sdk:

```ts
// gc-test.ts
it.only('gc-test', async function ({
  task,
}) {
  type TestDoc = { point?: { x?: number } };
  const docKey = toDocKey(`${task.name}-${new Date().getTime()}`);
  const doc1 = new yorkie.Document<TestDoc>(docKey);
  const doc2 = new yorkie.Document<TestDoc>(docKey);

  const client1 = new yorkie.Client(testRPCAddr);
  const client2 = new yorkie.Client(testRPCAddr);

  await client1.activate();
  await client2.activate();

  await client1.attach(doc1, { isRealtimeSync: false });
  doc1.update((root) => (root.point = { x: 0 }));
  await client1.sync();
  await client2.attach(doc2, { isRealtimeSync: false });

  doc1.update((root) => {
    delete root.point;
  });
  assert.equal(doc1.getGarbageLen(), 2);

  doc2.update((root) => {
    delete root.point?.x;
  });
  assert.equal(doc2.getGarbageLen(), 1);

  await client1.sync();
  await client2.sync();
  await client1.sync();

  // to-be: getGarbageLen() should also be 2.
  assert.equal(doc1.getGarbageLen(), 3);
  assert.equal(doc2.getGarbageLen(), 3);

  assert.equal(doc1.garbageCollect(MaxTimeTicket), 2);
  assert.equal(doc2.garbageCollect(MaxTimeTicket), 2);

  await client1.deactivate();
  await client2.deactivate();
});
```

**Anything else we need to know?**:

**Environment**:
- Operating system:
- Browser and version:
- Yorkie version (use `yorkie version`): v0.4.8
- Yorkie JS SDK version: v0.4.8

### PR DESCRIPTION
**Pull request by devleejb**
**What this PR does / why we need it**:

- To ensure that the `GarbageLen` function returns the correct length, I used a Map to eliminate duplicate counting
- Add test case for `GarbageLen` (same with the test case in [this PR](https://github.com/yorkie-team/yorkie-js-sdk/pull/714))

**Which issue(s) this PR fixes**:

Fixes #688 

**Special notes for your reviewer**:

Refer [yorkie-js-sdk/PR #714](https://github.com/yorkie-team/yorkie-js-sdk/pull/714)
The PR solves the same problem. And you can track the optimization history.

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation**:

```docs

```

**Checklist**:
- [x] Added relevant tests or not required
- [x] Didn't break anything

### CODE REVIEW

File: pkg/document/crdt/root.go [hunk 1]
```diff
@@ -151,19 +151,22 @@ func (r *Root) RemovedElementLen() int {
 // GarbageLen returns the count of removed elements.
 func (r *Root) GarbageLen() int {
 	count := 0
+	seen := make(map[string]bool)
 
 	for _, pair := range r.removedElementPairMapByCreatedAt {
-		count++
+		seen[pair.elem.CreatedAt().ToTestString()] = true
```
Comment (hackerwins) [inline pkg/document/crdt/root.go:157]: `ToTestString` is intended to return a value for testing purposes, so it does not seem appropriate to use in logic.

```suggestion
		seen[pair.elem.CreatedAt().Key()] = true
```

**Review by hackerwins — CHANGES_REQUESTED** [review]
Thanks for your contribution.
I left a comment.

File: pkg/document/crdt/root.go (hunk 1 above)
Comment (devleejb) [reply pkg/document/crdt/root.go]: Yes, the changes have been applied.

Actually, I was looking for a function that performs the same role as `toIDString` in `yorkie-js-sdk`, but couldn't find it. Could I know why the names differ between `toIDString` in `yorkie-js-sdk` and `Key` in `yorkie`?

**Review by hackerwins — APPROVED** [review]
LGTM. 👍

**Pull request merged**